### PR TITLE
Enable strip mining decisions

### PR DESCRIPTION
**Pull request by ulysseB**
This PR enable strip mining decisions in Telamon. Most of the work on the constraints and the model was done before. This PR:
- changes `ir::Size` to depend on the size decisions exposed in the domain,
- updates `mode::size` and `codegen::size` to retrieve the value of an `ir::Size` from the domain,
- updates the builder to expose tiling decisions through a `TilingPattern` structure, and
- updates the kernels to use tiling decisions.

Fixes #131

### CODE REVIEW

File: kernels/benches/cuda_search.rs [hunk 1]
```diff
@@ -48,7 +48,10 @@ fn main() {
         matmul_reference(&cublas_handle, params, ctx)
     });
     // 0.87 in 2.38 hours/4H
-    let params = linalg::MatMulP::new(1024, 1024, 1024).static_sizes();
+    let mut params = linalg::MatMulP::new(1024, 1024, 1024);
```
Comment (Elarnon) [inline kernels/benches/cuda_search.rs]: Should we have a builder helper for this?

**Comment by ulysseB** [comment]
I added an explanation for the TilingPattern.

File: kernels/src/lib.rs [hunk 2]
```diff
@@ -22,26 +21,18 @@ pub mod statistics;
...
-    tile_sizes: Vec<Vec<u32>>,
 ) -> explorer::Candidate<'a> {
     let bound = model::bound(&space, ctx);
-    let mut cand = explorer::Candidate::new(space, bound);
-    let acts = cand
-        .actions
-        .push_front(explorer::choice::ActionEx::TileSizes(tile_sizes));
-    cand.actions = acts;
-    cand
+    explorer::Candidate::new(space, bound)
```
Comment (Elarnon) [inline kernels/src/lib.rs]: :sparkles:

File: src/device/x86/cpu.rs [hunk 3]
```diff
@@ -42,7 +42,7 @@ impl device::Device for Cpu {
     }
 
     fn max_inner_block_size(&self) -> u32 {
-        0
+        1
```
Comment (Elarnon) [inline src/device/x86/cpu.rs]: I guess I trust you on this one, but out of curiosity -- what does this change?

File: src/explorer/candidate.rs [hunk 4]
```diff
@@ -60,7 +60,6 @@ impl<'a> Candidate<'a> {
         debug!("applying action {:?}", action);
         let mut space = self.space.clone();
         match action {
-            ActionEx::TileSizes(..) => panic!(),
```
Comment (Elarnon) [inline src/explorer/candidate.rs]: :sparkles:

**Pull request merged**